internal/controller/userdeactivation_controller.go

```diff
@@ -0,0 +1,245 @@
+			return ctrl.Result{}, fmt.Errorf("failed to update User status: %w", err)
+		}
+	} else {
+		log.Info("User is already deactivated", "userRef", userDeactivation.Spec.UserRef)
+	}
+
+	log.Info("Updating UserDeactivation status", "userDeactivationName", userDeactivation.GetName())
+	userDeactivationCondition := metav1.Condition{
+		Type:               userDeactivationReadyCondition,
+		Status:             metav1.ConditionTrue,
+		Reason:             "Reconciled",
+		Message:            "UserDeactivation successfully reconciled",
+		LastTransitionTime: metav1.Now(),
+	}
+	meta.SetStatusCondition(&user.Status.Conditions, userDeactivationCondition)
+	if err := r.Client.Status().Update(ctx, userDeactivation); err != nil {
+		log.Error(err, "Failed to update UserDeactivation status")
+		return ctrl.Result{}, fmt.Errorf("failed to update UserDeactivation status: %w", err)
+	}
+
+	log.Info("Reconciliation completed", "userRef", userDeactivation.Spec.UserRef)
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *UserDeactivationController) SetupWithManager(mgr mcmanager.Manager) error {
+	r.Finalizers = finalizer.NewFinalizers()
+	if err := r.Finalizers.Register(userDeactivationFinalizerKey, &userDeactivationFinalizer{
+		Client:  r.Client,
+		Zitadel: r.Zitadel,
+	}); err != nil {
+		return fmt.Errorf("failed to register group finalizer: %w", err)
+	}
+
+	return mcbuilder.ControllerManagedBy(mgr).
+		For(&iammiloapiscomv1alpha1.UserDeactivation{}).
+		Named("userdeactivation").
+		Complete(r)
+}
+
+// countOtherUserDeactivations counts how many other UserDeactivation objects
+// reference the same user, excluding the current UserDeactivation object
+func (f *userDeactivationFinalizer) countOtherUserDeactivations(ctx context.Context, userDeactivation *iammiloapiscomv1alpha1.UserDeactivation, userRef string) (int, error) {
+	log := logf.FromContext(ctx).WithName("count-other-deactivations")
+
+	// List all UserDeactivation objects to see if any other ones reference the same user
+	userDeactivationList := &iammiloapiscomv1alpha1.UserDeactivationList{}
+	err := f.Client.List(ctx, userDeactivationList)
+	if err != nil {
+		log.Error(err, "Failed to list UserDeactivation objects")
+		return 0, fmt.Errorf("failed to list UserDeactivation objects: %w", err)
+	}
+
+	// Count how many UserDeactivation objects reference the same user (excluding the current one)
+	otherDeactivationsCount := 0
+	for _, item := range userDeactivationList.Items {
+		// Skip the current UserDeactivation object
+		if item.GetUID() == userDeactivation.GetUID() {
+			continue
+		}
+		// Check if this UserDeactivation references the same user
+		if item.Spec.UserRef.Name == userRef {
+			otherDeactivationsCount++
+		}
+	}
+
+	log.Info("Other UserDeactivation objects found", "userRef", userRef, "count", otherDeactivationsCount)
+	return otherDeactivationsCount, nil
+}
```

Comment:
We should add some protections to Milo to ensure only one UserDeactivation resource exists per User.